### PR TITLE
Remove momentjs

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR removes momentjs and drops deprecated methods from core-utils. There is a breaking type change in the types package. Other packages are refactored to accommodate removal of moment or deprecated methods. The PR should trigger a major release to core-utils and types, and patch versions for other packages.

Fix https://github.com/opentripplanner/otp-ui/issues/359 by determining what constitutes "today", "tomorrow" or "yesterday" by comparing calendar days instead of 24-hour periods in the AlertsBody.

### CODE REVIEW

**Review by miles-grant-ibigroup — APPROVED** [review]
This looks great! Thank you very much for removing those deprecated core-utils methods. Very clean and everything works as expected!

otp-rr will need to be updated, but it looks like the changes won't be massive. Also, good luck with removing the alpha dependencies...

File: __snapshots__/storybook.test.ts.snap [hunk 1]
```diff
@@ -94662,7 +94663,8 @@ exports[`Storyshots LocationField/Mobile Context With Custom Icons 2`] = `
           <div
             className="c18"
           >
-            Home (456 Suburb St)
+            Home
+             (456 Suburb St)
             <span
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply __snapshots__/storybook.test.ts.snap:94668]: `<FormattedMessage>` renders into a React Fragment, and it seems that for snapshots, the contents of each fragment or react element is placed into a new line.

File: packages/location-field/src/options.tsx [hunk 2]
```diff
@@ -115,3 +115,59 @@ export function UserLocationIcon({
...
+      <FormattedMessage
+        defaultMessage="Work"
+        description="The work location"
+        id="otpUi.LocationField.workLocation"
+      />
+    );
+  } else {
+    displayName = location.name;
+  }
+
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/location-field/src/options.tsx]: Sure, updated in https://github.com/opentripplanner/otp-ui/pull/428/commits/82fd468595868190e31d694542e65f9568228614.

File: packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx [hunk 3]
```diff
@@ -42,9 +42,7 @@ if (!isRunningJest()) {
   const now = todayWithTime.valueOf();
   alerts[0].effectiveStartDate = now; // Today
   alerts[1].effectiveStartDate = now - 24 * 3600000; // Yesterday
-  // FIXME: Fix the criterion that decides what constitutes "tomorrow"
-  // (Adding 24 hours so that a timestamp occurs next day may not be enough).
-  alerts[2].effectiveStartDate = now + 36 * 3600000; // Tomorrow
+  alerts[2].effectiveStartDate = now + 24 * 3600000 + 10000; // Tomorrow
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/stories/OtpRrItineraryBody.story.tsx]: No, thanks, it is the remnant of an unnecessary hack. Removed in https://github.com/opentripplanner/otp-ui/pull/428/commits/82fd468595868190e31d694542e65f9568228614.

File: packages/itinerary-body/src/demos/index.tsx [hunk 4]
```diff
@@ -59,7 +58,7 @@ export function CustomTransitLegSummary({
     /* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */
     <div onClick={onClick}>
       {leg.duration && (
-        <span>Ride {coreUtils.time.formatDuration(leg.duration, "en-US")}</span>
+        <span>Ride for a custom duration of {leg.duration / 60} minutes</span>
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/demos/index.tsx]: Okay, I limited to 3 places in https://github.com/opentripplanner/otp-ui/pull/428/commits/82fd468595868190e31d694542e65f9568228614.

File: packages/core-utils/src/__tests__/itinerary.js [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
+  calculateTncFares,
   getCompanyFromLeg,
   getTransitFare,
   isTransit
 } from "../itinerary";
 
 const bikeRentalItinerary = require("./__mocks__/bike-rental-itinerary.json");
 const tncItinerary = require("./__mocks__/tnc-itinerary.json");
-const multiCurrencyItinerary = require("./__mocks__/multi-currency-itinerary.json");
+// const multiCurrencyItinerary = require("./__mocks__/multi-currency-itinerary.json");
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/core-utils/src/__tests__/itinerary.js]: Thanks, I lost track of it. Removed in https://github.com/opentripplanner/otp-ui/pull/428/commits/82fd468595868190e31d694542e65f9568228614.

File: packages/itinerary-body/src/demos/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (daniel-heppner-ibigroup) [reply packages/itinerary-body/src/demos/index.tsx]: A user almost never needs to see more than 1 decimal place of minutes, do you think 3 might be excessive?

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Great work! Can merge as is, but I had one little thought if you feel like it.

File: packages/core-utils/src/itinerary.ts [hunk 6]
```diff
@@ -445,12 +423,53 @@ export function calculatePhysicalActivity(
...
+export function getTransitFare(
+  fareComponent: Money
+): {
+  currencyCode: string;
+  transitFare: number;
+} {
+  // Default values (if fare component is not valid).
+  let transitFare = 0;
+  let currencyCode = "USD";
+  if (fareComponent) {
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/itinerary.ts:465]: In some parts of the UI we solve this by explicitly checking for null

File: packages/itinerary-body/src/demos/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/demos/index.tsx]: Yep, it is an illustrative custom component.

**Pull request merged**